.github/workflows/trigger-all.yml

```diff
@@ -0,0 +1,44 @@
+name: build and test jobs for nightly and release
+run-name: ${{ github.actor }} triggered nightly or release on ${{ github.ref }}
+on:
+    schedule:
+      # * is a special character in YAML so you have to quote this string
+      - cron: '30 0 * * *'  # nightly run
+
+    workflow_dispatch:
+        inputs:
+            wf_category:
+                description: "workflow category, default is NIGHTLY"
+                type: choice
+                options:
+                    - NIGHTLY
+                    - RELEASE
+                default: NIGHTLY
+            push_to_pypi:
+                description: "when set and tests pass, then '.whl' & '.tar.gz' will be pushed to public pypi"
+                type: boolean
+                default: true
+            gitref:
+                description: "git commit hash or tag name" 
+                type: string
+                default: 'main'
+
+    push:
+        branches:
+          - auto
+
+jobs:
+
+    BUILD-TEST:
+        uses: ./.github/workflows/build-test.yml
+        name: ${{ inputs.wf_category || 'RELEASE' }}
+        with:
+            wf_category: ${{ inputs.wf_category || 'RELEASE' }}
+            gitref: ${{ inputs.gitref || 'auto' }}
+            push_to_pypi: ${{ inputs.push_to_pypi || false }}
+            test_configs: '[{"python":"3.11.4","label":"ubuntu-22.04","timeout":"40"},
```

Comment:
why are these, 3.10.12 and 3.11.4 running without GPUs?

Reply:
This was adopted from the existing workflows. I guess we also want to test on CPUs?

@dsikka can you please comment here? Do we want to use only GPUs for the runners, or should be a mixture of GPUs and CPUs?

Reply:
Checked with @dsikka and we want to cover both CPUs and GPUs.